src/data/fintype/card_embedding.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2021 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import data.fintype.card
+import data.equiv.fin
+import data.nat.factorial
+import data.equiv.embedding
+
+/-!
+# Birthday Problem
+
+This file establishes the cardinality of `α ↪ β` in full generality.
+-/
+
+local notation `|` x `|` := finset.card x
+local notation `‖` x `‖` := fintype.card x
+
+open_locale nat
+
+namespace fintype
+
+-- the [fintype α] argument may seem redundant; it isn't!
```

Comment:
Wait, why?

Reply:
Because without it, `unique.fintype` is inferred in the lemma statement and it doesn't match against situations where another `fintype` instance is used.

Reply:
Ok. Could you just replace the comment with something that says this? e.g.
```
-- We need the separate `fintype α` instance as it contains data,
-- and may not match definitionally with the instance coming from `unique.fintype`.
```